codenames-web-services/src/main/java/com/codenames/controllers/UsersController.java

```diff
@@ -0,0 +1,51 @@
+package com.codenames.controllers;
+
+import com.codenames.dto.CredentialDto;
+import com.codenames.dto.SignUpDto;
+import com.codenames.dto.UserAuthDto;
+import com.codenames.dto.UserDto;
+import com.codenames.mapper.UserMapper;
+import com.codenames.provider.UserAuthProvider;
+import com.codenames.services.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@RequiredArgsConstructor
+public class UsersController {
+
+    private final UserService userService;
+
+    private final UserAuthProvider userAuthProvider;
+
+    private final UserMapper userMapper;
+
+    @PostMapping("/login")
+    public ResponseEntity<UserAuthDto> loginUser(@RequestBody CredentialDto credentialDto){
+        UserAuthDto user = userMapper.userEntityToUserAuthDto(userService.login(credentialDto));
+
+        user.setJwtToken(userAuthProvider.createToken(user.getLogin()));
+
+        return ResponseEntity.ok(user);
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<UserAuthDto> registerUser(@RequestBody SignUpDto signUpDto){
+        UserAuthDto user = userMapper.userEntityToUserAuthDto(userService.register(signUpDto));
+
+        user.setJwtToken(userAuthProvider.createToken(user.getLogin()));
+
+        return ResponseEntity.ok(user);
+    }
+
+    @GetMapping("/check_user_is_auth")
+    public boolean checkUserIsAuth(){
+        return true;
```

Comment:
What is the purpose of this method and why it just returns true?

Reply:
This method is intended only to find out if the user is authorized, that is, when a request comes to it, if the user is not authorized, an error will be returned, and if true, the user is authorized. This is the only thing it is intended for.

Reply:
Generally the accepted way to check this is to have /whoami endpoint which returns current user (or guest if that's the expected role), and check if the user is authorized based on that

Reply:
I have slightly modified this method, I think after adding the division into user roles I will finalize it